vite.plugins.ts

```diff
@@ -79,3 +81,60 @@ export const minifyHTML = (): {
     return minify(html, { collapseWhitespace: true });
   },
 });
+
+/**
+ * Forwards requests to the local replica.
+ * Denies access to raw URLs.
+ *
+ * @param replicaOrigin Replica URL to forward requests to
+ * @param forwardRules List of rules (i.e. hostname to canisterId mappings)
+ *                     to forward requests to a specific canister
+ */
+export const replicaForwardPlugin = (
+  replicaOrigin: string,
+  forwardRules: { hosts: string[]; canisterId: string }[]
```

Comment:
_nitpick/subjectivity alert_

```suggestion
  forwardRules: Array<{ canisterId: string; hosts: string[]; }>
```

I missed the `[]` at the end, using `Array<>` makes it clearer; plus starting with `canisterId` makes it more consistent with other maps (like `Record<Foo,Bar>`) since it's basically used as a lookup